justfile

```diff
@@ -0,0 +1,155 @@
+export RUSTFLAGS := "-D warnings"
+export RUSTDOCFLAGS := "-D warnings"
+
+
+
+[private]
+default:
+	just -l --unsorted
+
+
+
+# Install rust 1.65, cargo-deny, cargo-fuzz, cargo-hack, and cargo-semver-checks
+install: install_rust_1_65 \
+         install_cargo_deny \
+	 install_cargo_fuzz \
+	 install_cargo_hack \
+	 install_cargo_semver_checks 
+
+[private]
+install_rust_1_65:
+	rustup toolchain install 1.65 --profile minimal
+
+[private]
+install_cargo_deny:
+	cargo install --locked cargo-deny
+ 
+[private]
+install_cargo_fuzz:
+	cargo install cargo-fuzz
+
+[private]
+install_cargo_hack:
+	cargo install --locked cargo-hack
+
+[private]
+install_cargo_semver_checks:
+	cargo install --locked cargo-semver-checks
+
+
+
+# Check syntax, formatting, clippy, deny, ...
+quick: (_quick ""               ""         ) \
+       (_quick ""               "--release") \
+       (_quick "--all-features" ""         ) \
+       (_quick "--all-features" "--release")
+
+_quick features mode: (cargo_check features mode) cargo_fmt (cargo_clippy features mode) cargo_deny
```

Comment:
Maybe you can rewrite it to `[private] quick_impl` instead of using the old private method naming convention with `_`. It might be less confusing for beginners (like me).